dynamo/preprocessing/preprocessor_utils.py

```diff
@@ -1210,34 +1210,44 @@ def calc_sz_factor(
+    This function is partly based on Monocle R package
+    (https://github.com/cole-trapnell-lab/monocle3).
 
     Args:
         adata_ori: an AnnData object.
-        layers: the layer(s) to be normalized. Defaults to "all", including RNA (X, raw) or spliced, unspliced, protein,
-            etc.
-        total_layers: the layer(s) that can be summed up to get the total mRNA. for example, ["spliced", "unspliced"],
-            ["uu", "ul", "su", "sl"] or ["new", "old"], etc. Defaults to None.
-        splicing_total_layers: whether to also normalize spliced / unspliced layers by size factor from total RNA.
-            Defaults to False.
-        X_total_layers: whether to also normalize adata.X by size factor from total RNA. Defaults to False.
+        layers: the layer(s) to be normalized. Defaults to "all", including RNA
+            (X, raw) or spliced, unspliced, protein, etc.
+        total_layers: the layer(s) that can be summed up to get the total mRNA.
+            For example, ["spliced", "unspliced"], ["uu", "ul", "su", "sl"] or
+            ["new", "old"], etc. Defaults to None.
+        splicing_total_layers: whether to also normalize spliced / unspliced
+            layers by size factor from total RNA. Defaults to False.
+        X_total_layers: whether to also normalize adata.X by size factor from
+            total RNA. Defaults to False.
         locfunc: the function to normalize the data. Defaults to np.nanmean.
-        round_exprs: whether the gene expression should be rounded into integers. Defaults to False.
-        method: the method used to calculate the expected total reads / UMI used in size factor calculation. Only
-            `mean-geometric-mean-total` / `geometric` and `median` are supported. When `median` is used, `locfunc` will
-            be replaced with `np.nanmedian`. Defaults to "median".
-        scale_to: the final total expression for each cell that will be scaled to. Defaults to None.
-        use_all_genes_cells: whether all cells and genes should be used for the size factor calculation. Defaults to
-            True.
-        genes_use_for_norm: A list of gene names that will be used to calculate total RNA for each cell and then the
-            size factor for normalization. This is often very useful when you want to use only the host genes to
-            normalize the dataset in a virus infection experiment (i.e. CMV or SARS-CoV-2 infection). Defaults to None.
+        round_exprs: whether the gene expression should be rounded into
+            integers. Defaults to False.
+        method: the method used to calculate the expected total reads / UMI used
+            in size factor calculation. Only `mean-geometric-mean-total` /
+            `geometric` and `median` are supported. When `median` is used,
+            `locfunc` will be replaced with `np.nanmedian`. Defaults to
+            "median".
```

Comment:
explain the difference between `mean-geometric-mean-total`, `geometric` and `median`